teamsyncd/teamsync.cpp

```diff
@@ -194,6 +215,41 @@ void TeamSync::removeLag(const string &lagName)
     m_selectablesToRemove.insert(lagName);
 }
 
+#define MAX_PID_STRLEN 10
+void TeamSync::cleanTeamProcesses(int signo)
+{
+    ssize_t sz = 0;
+    int fd, pid;
+    char pid_str[MAX_PID_STRLEN] = {0};
+    char *endptr = NULL;
+    string file_path;
+
+    for (auto it=m_teamSelectables.begin(); it!=m_teamSelectables.end(); ++it)
+    {
+       /* There are PID files created for each teamd process in /var/run/teamd DIR */
+       file_path = string(PID_FILE_PATH) + it->first + string(".pid");
+
+       /* Open the file, get the PID and send signal to the process */
+       fd = open(file_path.c_str(), O_RDONLY);
```

Comment:
probably it's easier to use c++ streams here.

Reply:
Plan was to use the signal safe library calls ( as in Linux Programmer's Manual ) to the maximum extend.